### ISSUE TITLE
Guides for block level customization

### ISSUE THREAD

**Issue by yl4070** [FluxML/FastAI.jl#183]
Hi,
Currently, the customization tutorial is based on low level API, it would be nice to have a similar guides but based on block level interface, covering materials such as how to write customized blocks for different use cases, and what functions need to be implemented for the new block to work properly. It appears to be a big jump from using existing blocks to customize everything. 

Personally, I'm struggling to figure out how to make new blocks work properly by looking at the source code, so I'm really looking forward to such guides.

**Comment by lorenzoh** [comment]
Yeah, documentation for that hasn't been written yet. If you want to implement new `Encoding`s for existing blocks, for now the best resource is [the docstring for `Encoding`](https://github.com/FluxML/FastAI.jl/blob/3a017af061a1125231fe7d7a4ec98da0a255d781/src/datablock/encoding.jl#L2). You can also look at and use `FastAI.testencoding` as a reference for the encoding being implemented properly. If you have any specific questions or stumbling blocks,  feel free to ask, though!

I'll leave this open and give an update once a proper guide is in the docs.
